src/emit/substrate/target.rs

```diff
@@ -1520,23 +1516,13 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
             byte_ptr!().const_null()
         };
 
-        let (data_ptr, data_len) = self.abi_encode(
-            binary,
-            event_id(binary, contract, event_no),
-            false,
-            function,
-            data,
-            data_tys,
-            ns,
-        );
-
         call!(
             "seal_deposit_event",
             &[
                 topic_buf.into(),
                 topic_size.into(),
-                data_ptr.into(),
-                data_len.into(),
+                binary.vector_bytes(data).into(),
+                size.into_int_value().into(),
```

Comment:
This could be `binary.vector_len(data).into()`. I am not sure why the length is passed in separately.

Reply:
In the solana emit target, the data and len were passed as different elements in a vector, I think thats why I did it just like that.